### PR TITLE
WIP Darigaaz, Shivan Champion

### PR DESCRIPTION
**Pull request by Simisays**
Needed to add params to MakeCardEffect but haven't found succes so far. Am I close to getting this working ? 
image

### CODE REVIEW

File: forge-game/src/main/java/forge/game/ability/effects/MakeCardEffect.java [hunk 1]
```diff
@@ -128,4 +147,4 @@ public void resolve(SpellAbility sa) {
             }
         }
     }
-}
+}
```
Comment (Northmoc) [inline forge-game/src/main/java/forge/game/ability/effects/MakeCardEffect.java:150]: please restore the newline at the end of the file

File: forge-gui/res/cardsfolder/upcoming/darigaaz_shivan_champion.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Name:Daringaaz, Shivan Dragon
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/darigaaz_shivan_champion.txt:1]: spell the name correctly

File: forge-gui/res/cardsfolder/upcoming/darigaaz_shivan_champion.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ManaCost:2 B R G
+Types:Legendary Creature Dragon
+PT:5/5
+K:Flying
+T:Mode$ Phase | Phase$ End of Turn | Execute$ TrigDraft | ValidPlayer$ You | TriggerZones$ Battlefield | TriggerDescription$ At the beginning of your end step, conjure a random card from CARDNAME's spellbook into exile face down with three egg counters on it. It gains "at the beginning of your upkeep, if this card is exiled, remove an egg counter from it. Then if it has no egg counters on it, put it onto the battlefield.  
+SVar:TrigDraft:DB$ MakeCard | Conjure$ True | AtRandom$ True | Spellbook$ Shivan Dragon,Moonveil Regent,Terror of the Peaks,Leyline Tyrant,Immersturm Predator,Manaform Hellkite,Bone Dragon,Demanding Dragon,Skarrgan Hellkite, Thunderbreak Regent,Black Dragon,Skyship Stalker,Red Dragon | WithCountersType$ EGG | WithCountersAmount$ 3 | Zone$ Exile | ExileFaceDown$ True | RememberDrafted$ True | SubAbility$ DBEffect
+SVar:DBEffect:DB$ Effect | RememberObjects$ Remembered | StaticAbilities$ RemoveEggCounter | Duration$ Permanent | Name$ Daringaaz, Shivan Dragon's Perpetual Effect
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/darigaaz_shivan_champion.txt:8]: I think this should be an Animate, not an Effect

File: forge-gui/res/cardsfolder/upcoming/darigaaz_shivan_champion.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+Name:Daringaaz, Shivan Dragon
+ManaCost:2 B R G
+Types:Legendary Creature Dragon
+PT:5/5
+K:Flying
+T:Mode$ Phase | Phase$ End of Turn | Execute$ TrigDraft | ValidPlayer$ You | TriggerZones$ Battlefield | TriggerDescription$ At the beginning of your end step, conjure a random card from CARDNAME's spellbook into exile face down with three egg counters on it. It gains "at the beginning of your upkeep, if this card is exiled, remove an egg counter from it. Then if it has no egg counters on it, put it onto the battlefield.  
+SVar:TrigDraft:DB$ MakeCard | Conjure$ True | AtRandom$ True | Spellbook$ Shivan Dragon,Moonveil Regent,Terror of the Peaks,Leyline Tyrant,Immersturm Predator,Manaform Hellkite,Bone Dragon,Demanding Dragon,Skarrgan Hellkite, Thunderbreak Regent,Black Dragon,Skyship Stalker,Red Dragon | WithCountersType$ EGG | WithCountersAmount$ 3 | Zone$ Exile | ExileFaceDown$ True | RememberDrafted$ True | SubAbility$ DBEffect
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/darigaaz_shivan_champion.txt:7]: There is no `RememberDrafted` – use `RememberMade`

File: forge-gui/res/cardsfolder/upcoming/darigaaz_shivan_champion.txt [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+SVar:DBEffect:DB$ Effect | RememberObjects$ Remembered | StaticAbilities$ RemoveEggCounter | Duration$ Permanent | Name$ Daringaaz, Shivan Dragon's Perpetual Effect
+SVar:RemoveEggCounter:Mode$ Continuous | Affected$ Card.IsRemembered | AddTrigger$ TrigRemoveEgg | EffectZone$ Command | AffectedZone$ Battlefield,Hand,Graveyard,Exile,Stack,Library,Command | Description$ at the beginning of your upkeep, if this card is exiled, remove an egg counter from it. Then if it has no egg counters on it, put it onto the battlefield. 
+SVar:TrigRemoveEgg:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Exile | IsPresent$ Card.IsRemembered | PresentZone$ Exile | Execute$ DBRemoveCounter | TriggerDescription$ At the beginning of your upkeep, if CARDNAME is exiled, remove an egg counter from it. Then if CARDNAME has no egg counters on it, return it to the battlefield.
+SVar:DBRemoveCounter:DB$ RemoveCounter | Defined$ Remembered | CounterType$ EGG | CounterNum$ 1 | SubAbility$ DBReturn
+SVar:DBReturn:DB$ ChangeZone | Defined$ Remembered | Origin$ Exile | Destination$ Battlefield | ConditionDefined$ Remembered | ConditionPresent$ Card.counters_EQ0_EGG
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/darigaaz_shivan_champion.txt:12]: This need a little tweaking once you use Animate

File: forge-gui/res/cardsfolder/upcoming/darigaaz_shivan_champion.txt (hunk 3 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/darigaaz_shivan_champion.txt:8]: Good point, though generally the cardscript is very rough. I'm having trouble adding the param for counters when you exile the made card

**Pull request closed without merging**